client/state/reader/lists/reducer.js

```diff
@@ -32,17 +34,13 @@ import { itemsSchema, subscriptionsSchema } from './schema';
 export const items = withSchemaValidation( itemsSchema, ( state = {}, action ) => {
 	switch ( action.type ) {
 		case READER_LISTS_RECEIVE:
-			return Object.assign( {}, state, keyBy( action.lists, 'ID' ) );
-		case READER_LIST_REQUEST_SUCCESS:
-		case READER_LIST_UPDATE_SUCCESS:
-			return Object.assign( {}, state, keyBy( [ action.data.list ], 'ID' ) );
-		case READER_LIST_DELETE:
-			if ( ! ( action.listId in state ) ) {
-				return state;
-			}
-			return omit( state, action.listId );
```

Comment:
Do we no longer need these?

Reply:
I think there might be old, legacy `lists` functionality in here confounding the situation.

Reply:
This change isn't necessary for this PR and deals with existing functionality, so I reverted it.
As we continue working on the Lists feature, we can determine later whether it makes sense to refactor.